### PR TITLE
feat: add popover (aka flyout) component

### PR DESCRIPTION
**Pull request by SethDonohue**
# Description

Adds a Popover behavior component.

## Motivation & context

There is a use case for having a behavior that allows for content to be displayed in front of current content to provide additional content or actions and tied to some other element. Similar to a Dialog, but it won't be a modal, or force the user to make a decision on something, Popover will have engage the user and require the user to at least close the popover if they no longer want to view that content or provide the requested information, etc.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [x] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: packages/web-components/fast-foundation/src/popover/popover.ts [hunk 1]
```diff
@@ -0,0 +1,634 @@
...
+
+export { PopoverPosition };
+
+/**
+ * An Popover Custom HTML Element.
+ *
+ * @public
+ */
+export class Popover extends FASTElement {
+    private static DirectionAttributeName: string = "dir";
```
Comment (scomea) [inline packages/web-components/fast-foundation/src/popover/popover.ts:20]: I believe this is unused.

**Comment by scomea** [comment]
I think we need some mechanism that updates the position of the popover in case layout changes.  Tooltips don't do this but they are transient enough that it's not such a big deal, popover is meant to contain active focusable ui and should be able to remain usable after resizing/scrolling so people will notice.  I think we should consider adding a timer that calls update() on the underlying anchored region a few times a second so that it automatically corrects layout after scrolling/resizing.  Probably an option to turn that on/off as well.   I have a working branch that does this for tooltip, including animating to the updated position.  https://github.com/microsoft/fast/tree/users/scomea/tooltip-track

**Comment by SethDonohue** [comment]
> I think we need some mechanism that updates the position of the popover in case layout changes. Tooltips don't do this but they are transient enough that it's not such a big deal, popover is meant to contain active focusable ui and should be able to remain usable after resizing/scrolling so people will notice. I think we should consider adding a timer that calls update() on the underlying anchored region a few times a second so that it automatically corrects layout after scrolling/resizing. Probably an option to turn that on/off as well. I have a working branch that does this for tooltip, including animating to the updated position. https://github.com/microsoft/fast/tree/users/scomea/tooltip-track

Probably an excellent idea. I'll take a look at your branch when finishing this implementation. 
Do you think I should add something to the spec (https://github.com/microsoft/fast/pull/3691) before completing the spec, or add it after we get this component mostly sorted?

File: packages/web-components/fast-foundation/src/popover/popover.ts (hunk 1 above)
Comment (SethDonohue) [reply packages/web-components/fast-foundation/src/popover/popover.ts:20]: Likely, I haven't finished this just yet, but will take a look at this.

**Comment by scomea** [comment]
> Do you think I should add something to the spec (#3691) before completing the spec, or add it after we get this component mostly sorted?

I think it's something we should address before we ship it, but purists would want to see the spec first.

**Comment by scomea** [comment]
Another "spec" question - fixed vs. absolute positioning.  Fixed breaks out of containers but has latency on scrolling and updates needed to track the anchor. You currently have absolute positioning - is that the right choice, should developers be able to choose?

**Comment by SethDonohue** [comment]
> Another "spec" question - fixed vs. absolute positioning. Fixed breaks out of containers but has latency on scrolling and updates needed to track the anchor. You currently have absolute positioning - is that the right choice, should developers be able to choose?

I'll have to look in to this further. If I recall correctly I couldn't get this to work without absolute positioning, but you just added a recent change that works with flex on anchored region, correct?

**Pull request closed without merging**